src/T2DMap.cpp

```diff
@@ -1923,6 +1958,63 @@ void T2DMap::paintAreaExits(QPainter& painter, QPen& pen, QList<int>& exitList,
                     }
                     painter.drawPolyline(polyLinePoints.data(), polyLinePoints.size());
 
+                    if (!getUserDataBool(room->userData, ROOM_UI_DONTSHOWAREAEXITNAME, false)) {
+                        QPointF areaExitNameOffset = parseOffsetString(room->userData.value(ROOM_UI_AREAEXITNAMEOFFSET + "." + itk.key()));
+                        int exitRoomId = -1;
+                        QMapIterator<int, QString> otherExitIt = room->getOtherMap();
```

Comment:
:information_source: Since #4526 the way special exits (and special exit locks) have been stored more straightforwardly in the `TRoom` class - you are probably going to want to iterate through the return from `room->getSpecialExits()` now - which is a `const QMap<int, QString>&` (and won't have to have a leading `0` or `1` trimmed off that used to indicate the lock state from the name/command `QString`)...

Reply:
OK, fix it